rust/worker/src/execution/orchestration/compact.rs

```diff
@@ -323,139 +367,127 @@ impl CompactOrchestrator {
+        let hnsw_segment = self.get_segment(SegmentType::HnswDistributed).await?;
+
+        let borrowed_writers = self
+            .writers
+            .get_or_try_init::<GetSegmentWritersError, _, _>(|| async {
+                // Create a record segment writer
+                let record_segment_writer =
+                    match RecordSegmentWriter::from_segment(&record_segment, &blockfile_provider)
+                        .await
+                    {
+                        Ok(writer) => writer,
+                        Err(e) => {
+                            tracing::error!("Error creating Record Segment Writer: {:?}", e);
+                            return Err(GetSegmentWritersError::RecordSegmentWriterError);
+                        }
+                    };
+
+                tracing::debug!("Record Segment Writer created");
+
+                // Create a record segment writer
+                let mt_segment_writer =
+                    match MetadataSegmentWriter::from_segment(&mt_segment, &blockfile_provider)
+                        .await
+                    {
+                        Ok(writer) => writer,
+                        Err(e) => {
+                            println!("Error creating metadata Segment Writer: {:?}", e);
+                            return Err(GetSegmentWritersError::MetadataSegmentWriterError);
+                        }
+                    };
+
+                tracing::debug!("Metadata Segment Writer created");
+
+                // Create a hnsw segment writer
+                let collection_res = sysdb
+                    .get_collections(Some(self.collection_id), None, None, None)
+                    .await;
 
-        let collection_res = match collection_res {
-            Ok(collections) => {
-                if collections.is_empty() {
-                    return Err(Box::new(GetSegmentWritersError::CollectionNotFound));
+                let collection_res = match collection_res {
+                    Ok(collections) => {
+                        if collections.is_empty() {
+                            return Err(GetSegmentWritersError::CollectionNotFound);
+                        }
+                        collections
+                    }
+                    Err(e) => {
+                        return Err(GetSegmentWritersError::GetCollectionError(e));
+                    }
+                };
+                let collection = &collection_res[0];
+
+                if let Some(dimension) = collection.dimension {
+                    let hnsw_segment_writer = match DistributedHNSWSegmentWriter::from_segment(
+                        &hnsw_segment,
+                        dimension as usize,
+                        hnsw_provider,
+                    )
+                    .await
+                    {
+                        Ok(writer) => writer,
+                        Err(e) => {
+                            println!("Error creating HNSW Segment Writer: {:?}", e);
```

Comment:
can we remove prints in favor of trace/log please!

Reply:
whoops yes